tools/benchmark/src/mocha/runner.ts

```diff
@@ -14,8 +14,29 @@ import {
 	performanceTestSuiteTag,
 	userCategoriesSplitter,
 	TestType,
-} from "./Configuration";
-import { runBenchmark } from "./runBenchmark";
+	Titled,
+	MochaExclusiveOptions,
+	BenchmarkDescription,
+	BenchmarkTimingOptions,
+} from "../Configuration";
+import {
+	BenchmarkResult,
+	CustomBenchmark,
+	Phase,
+	runBenchmark,
+	runCustomBenchmark,
+} from "../runBenchmark";
+
+export function qualifiedTitle(args: BenchmarkDescription & Titled): string {
+	const benchmarkTypeTag = BenchmarkType[args.type ?? BenchmarkType.Measurement];
+	const testTypeTag = TestType[TestType.ExecutionTime];
+	let qualifiedTitle = `${performanceTestSuiteTag} @${benchmarkTypeTag} @${testTypeTag} ${args.title}`;
+
+	if (args.category !== "") {
```

Comment:
>(args.category [](http://example.com/codeflow?start=4&length=14)

in case category is undefined do we want to set qualifiedTitle as well ?

Reply:
we already do, two lines above this. Note that this policy and logic is unchanged, just moved.

Reply:
yes, I understood it was unchanged - on our PerfE2E I realized we do not set category and that codepath gets triggered with undefined as category :) That is why I mentioned it.